### PR TITLE
[Validator] Treat service blocks as struct blocks for ease of traversal

### PR DESCRIPTION
**Pull request by c-harding**
Just a quick one, should make dealing with both struct blocks and service blocks easier. Main result is that `structIterator` will return structs, whose metadata can be read, instead of simple maps of names to attributes

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
Cool cool cool

File: src/main/scala/temple/ast/OldServiceBlock.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package temple.ast
+
+import temple.ast.Metadata.ServiceMetadata
+
+/** A service block, representing one microservice on its own isolated server */
+case class OldServiceBlock(
```
Comment (jaylees14) [inline src/main/scala/temple/ast/OldServiceBlock.scala]: I can't see where this is used? Also old vs what?

File: src/main/scala/temple/ast/StructBlock.scala [hunk 2]
```diff
@@ -2,8 +2,7 @@ package temple.ast
 
 import temple.ast.Metadata.StructMetadata
 
-/** A nested struct within a service, representing a second table/domain in the same datastore */
-case class StructBlock(
-  attributes: Map[String, Attribute],
-  metadata: Seq[StructMetadata] = Nil,
-) extends TempleBlock[StructMetadata]
+trait StructBlock[M >: StructMetadata <: Metadata] extends TempleBlock[M] {
```
Comment (jaylees14) [inline src/main/scala/temple/ast/StructBlock.scala]: Hmm I'm not sure about the naming `Struct` and `NestedStruct`. I feel like `Struct` needs to stay as is (to match Templefile syntax) and this becomes something `AbstractStruct` to make it clear it's a trait? Not sure about names though

**Review by jaylees14 — APPROVED** [review]
Couple of suggestions, but happy to go ahead if you don't agree

File: src/main/scala/temple/ast/OldServiceBlock.scala (hunk 1 above)
Comment (c-harding) [reply src/main/scala/temple/ast/OldServiceBlock.scala]: Sorry, this should have been deleted following a merge conflict

**Pull request merged**